node/src/reactor/main_reactor/control.rs

```diff
@@ -93,7 +93,9 @@ impl MainReactor {
                 }
                 CatchUpInstruction::ShutdownForUpgrade => {
                     info!("CatchUp: shutting down for upgrade");
-                    self.state = ReactorState::ShutdownForUpgrade;
+                    self.state = ReactorState::ShutdownForUpgrade {
+                        time_switched: Timestamp::now(),
+                    };
```

Comment:
It might be a little bit less error-prone if we add `ReactorState::new_shutdown_for_upgrade()` and set the timestamp to `now()` internally.